modules/plot/src/plot.cpp

```diff
@@ -0,0 +1,324 @@
+    {
+        MinXFindZero = XAxisMin;
+        MaxXFindZero = XAxisMax;
+        MinYFindZero = YAxisMin;
+        MaxYFindZero = YAxisMax;
+    }
+
+    Mat InterpXdataFindZero = linearInterpolation(MinXFindZero, MaxXFindZero, 0, FigW, XdataPlusZero);
+    Mat InterpYdataFindZero = linearInterpolation(MaxYFindZero, MinYFindZero, 0, FigH, YdataPlusZero);
+
+    double ImageXzero = InterpXdataFindZero.at<double>(NumVecElements,0);
+    double ImageYzero = InterpYdataFindZero.at<double>(NumVecElements,0);
+
+    double CurrentX = Xdata.at<double>(NumVecElements-1,0);
+    double CurrentY = Ydata.at<double>(NumVecElements-1,0);
+
+    Mat FigureRed	= Mat::zeros(FigH, FigW, CV_64F);
+    Mat FigureGreen = Mat::zeros(FigH, FigW, CV_64F);
+    Mat FigureBlue	= Mat::zeros(FigH, FigW, CV_64F);
+    std::vector<cv::Mat> FigureVec;
+    Mat Figure		= Mat::zeros(FigH, FigW, CV_64F);
+
+    //Draw the plot by connecting lines between the points
+    cv::Point p1;
+    p1.x = InterpXdata.at<double>(0,0);
+    p1.y = InterpYdata.at<double>(0,0);
+
+    drawAxis(MinX, MaxX, MinY, MaxY, ImageXzero,ImageYzero, CurrentX, CurrentY, NumVecElements, FigW, FigH, FigureRed, FigureGreen, FigureBlue);
+
+    for (int r=1; r<InterpXdata.rows; r++){
+
+        cv::Point p2;
+        p2.x = InterpXdata.at<double>(r,0);
+        p2.y = InterpYdata.at<double>(r,0);
+
+        line(FigureRed, p1, p2, 255, 2, 8, 0);
+        line(FigureGreen, p1, p2, 255, 2, 8, 0);
+        line(FigureBlue, p1, p2, 0, 2, 8, 0);
+
+        p1 = p2;
+
+    }
+
+    FigureVec.push_back(FigureBlue);
+    FigureVec.push_back(FigureGreen);
+    FigureVec.push_back(FigureRed);
+
+    merge(FigureVec,Figure);
+
+    //Show the plot
+    namedWindow(FigureName);
+    imshow(FigureName, Figure);
+
+    waitKey(5);
+
+}
+
+void Plot::drawAxis(double /*MinX*/, double /*MaxX*/, double /*MinY*/, double /*MaxY*/, double ImageXzero, double ImageYzero, double CurrentX, double CurrentY, int /*NumVecElements*/, int FigW, int FigH, cv::Mat &FigureRed, cv::Mat &FigureGreen, cv::Mat &FigureBlue){
+
+    drawValuesAsText(0, ImageXzero, ImageYzero, 10, 20, FigureRed, FigureGreen, FigureBlue);
+    drawValuesAsText(0, ImageXzero, ImageYzero, -20, 20, FigureRed, FigureGreen, FigureBlue);
+    drawValuesAsText(0, ImageXzero, ImageYzero, 10, -10, FigureRed, FigureGreen, FigureBlue);
+    drawValuesAsText(0, ImageXzero, ImageYzero, -20, -10, FigureRed, FigureGreen, FigureBlue);
+    drawValuesAsText("X = %g",CurrentX, 0, 0, 40, 20, FigureRed, FigureGreen, FigureBlue);
+    drawValuesAsText("Y = %g",CurrentY, 0, 20, 40, 20, FigureRed, FigureGreen, FigureBlue);
+
+    //Horizontal X axis and equispaced horizontal lines
+    int LineSpace = 50;
+    int TraceSize = 5;
+    cv::Point3d RGBLineColor;
+    RGBLineColor.x = 255;
+    RGBLineColor.y = 0;
+    RGBLineColor.z = 0;
+    drawLine(0, FigW, ImageYzero, ImageYzero, FigureRed, FigureGreen, FigureBlue, 1, RGBLineColor);
```

Comment:
Same as above, the ImageYzero value should be defined in the function as int, not as double!